src/daemon/http/tls.rs

```diff
@@ -1,322 +1,380 @@
+        ///
+        /// Because hyper still wants to do a clean flush and shutdown, we
+        /// need to still work in this state. For read and write, we just
+        /// keep returning the clean shutdown indiciation of zero length
+        /// operations.
+        Empty,
     }
+}
 
-    /// Specify the file path for the TLS certificate to use.
-    pub(crate) fn cert_path(mut self, path: impl AsRef<Path>) -> Self {
-        self.cert = Box::new(LazyFile {
-            path: path.as_ref().into(),
-            file: None,
-        });
-        self
+impl TlsTcpStream {
+    fn new(sock: TcpStream, tls: &TlsAcceptor) -> Self {
+        Self::Accept { fut: tls.accept(sock) }
     }
 
-    pub(crate) fn build(mut self) -> Result<ServerConfig, TlsConfigError> {
-        let mut cert_rdr = BufReader::new(self.cert);
-        let certs = rustls_pemfile::certs(&mut cert_rdr).map_err(TlsConfigError::Io)?;
-        let cert = certs.into_iter().map(Certificate).collect();
-
-        let key = {
-            // convert it to Vec<u8> to allow reading it again if key is RSA
-            let mut key_vec = Vec::new();
-            self.key.read_to_end(&mut key_vec).map_err(TlsConfigError::Io)?;
-
-            if key_vec.is_empty() {
-                return Err(TlsConfigError::EmptyKey);
-            }
-
-            let mut pkcs8 = rustls_pemfile::pkcs8_private_keys(&mut key_vec.as_slice())
-                .map_err(|_| TlsConfigError::Pkcs8ParseError)?;
-
-            if !pkcs8.is_empty() {
-                pkcs8.remove(0)
-            } else {
-                let mut rsa = rustls_pemfile::rsa_private_keys(&mut key_vec.as_slice())
-                    .map_err(|_| TlsConfigError::RsaParseError)?;
-
-                if !rsa.is_empty() {
-                    rsa.remove(0)
-                } else {
-                    return Err(TlsConfigError::EmptyKey);
+    fn poll_accept(
+        mut self: Pin<&mut Self>,
+        cx: &mut Context<'_>,
+    ) -> Poll<Result<Pin<&mut Self>, io::Error>> {
+        match self.as_mut().project() {
+            TlsTcpStreamProj::Accept { fut } => {
+                match ready!(fut.try_poll(cx)) {
+                    Ok(fut) => {
+                        self.set(Self::Stream { fut });
+                        Poll::Ready(Ok(self))
+                    }
+                    Err(err) => {
+                        self.set(Self::Empty);
+                        Poll::Ready(Err(err))
+                    }
                 }
             }
-        };
-
-        let mut config = ServerConfig::builder()
-            .with_safe_default_cipher_suites()
-            .with_safe_default_kx_groups()
-            .with_safe_default_protocol_versions()
-            .map_err(TlsConfigError::Rustls)?
-            .with_no_client_auth()
-            .with_single_cert(cert, tokio_rustls::rustls::PrivateKey(key))
-            .map_err(TlsConfigError::Rustls)?;
-
-        // See: https://wiki.wireshark.org/TLS#tls-decryption
```

Comment:
Is there a reason that this comment wasn't carried over to the new code?

Reply:
Nope. I added it back in again.